### DISCUSSION TITLE
Export xml by ns

### DISCUSSION THREAD

**Discussion by roin93** [FreeOpcUa/opcua-asyncio#501] [Q&A]
Hello,

I know there is a function for the server to export xmls selected by namespaces (export_xml_by_ns()).
But is there the same functionality for the client?
I know there is a function as export_xml(path, nodes) but to use that in a proper way I would need to generate the whole list of nodes I want to select. Therefore I would need to iterate through EVERY node and check the namespace.

**Comment by roin93** [comment]
It would also help if there is a function like get_children_in_ns(). I did not find any. Is it possible to search the tree especially for a namespace without iterating through all children of all nodes?

**Reply by AndreasHeine** [reply]
no! its not part of the opc ua spec.

**Reply by AndreasHeine** [reply]
https://reference.opcfoundation.org/v104/
